### DISCUSSION TITLE
Error Message About Main.py on Bootup

### DISCUSSION THREAD

**Discussion by nitrobob** [leokhoa/laragon#700] [General]
When I booted my computer this morning, I received an unhandled exception script error that read "Failed to execute script 'main' due to unhandled exception: module 'x509" has no attribute 'Store'. The error message continued by saying File "main.py". Line 17, in ImportError: attempted relative import with no known parent package. It went on to list many additional exceptions.

I searched my system and only Laragon and Inkspace had a main.py file. I uninstalled Inkspace and updated Laragon the the latest version, but the error message appears every time I restart my computer. I am considering copying my C:\Laragon\www folder to a temporary location, uninstalling and reinstalling Laragon, and then restoring the www folder and testing to see if that resolves the issue. If anyone has a way to resolve this error message, I will appreciate your thoughts. I have attached the full error message.
[Error Message.txt](https://github.com/leokhoa/laragon/files/14378243/Error.Message.txt)

**Comment by nitrobob** [comment]
I went ahead and backed up my sites, uninstalled Laragon and rebooted.  Unfortunately, the error message reappeared.  I checked and I cannot locate main.py on my system after uninstalling Laragon, so this may not be related to Laragon after all.  I will continue to research the issue.

**Reply by Pen-y-Fan** [reply]
Laragon doesn't ship with Python, main.py is a Python script. 

Open Task Manager (CTRL + SHIFT + ESC) and check your **Startup apps** for anything suspicious. 

<img width="201" alt="image" src="https://github.com/leokhoa/laragon/assets/40126936/8f41ebc4-b825-446d-aa1f-0e24dfe4473e">

Run an auto-virus scan.

**Comment by nitrobob** [comment]
Pen-Y-Fan, thanks for your suggestions.  I was aware that main.py is a Python script.  I can confirm that when I just reinstalled Laragon, there is a laragon\bin\python\python-3.10 that is part of the installation.  I realize that it may not be directly part of Laragon, but that was the only location where I found a main.py file, other than Inkscape.  There are now nine such files in various subfolders of the Laragon folder.
I have checked the startups and found nothing unusual, and I am running a sequence of scanners to locate any potential malware.

**Reply by Charly-Bob** [reply]
Hi, I had the same error message. It seemed to be the app "devicehub" (Pitney Bowes) in my startup. After I deactivated it and rebooted the error message was gone...

**Reply by nitrobob** [reply]
Charly-Bob,  Thanks for this info.  I finally discovered this source of this issue when I booted up this morning.  I uninstalled DeviceHub, downloaded the latest version, and reinstalled.  When I opened Pitney Bowes SendPro, there was a new simplified user interface.  Since you also experienced this issue, my guess is that an auto-update may have contained a flaw.  DeviceHub apparently executes the initiation of the software strictly via the remote connection to the hosted service.
